tests/integration/test_session_management.py

```diff
@@ -0,0 +1,144 @@
+            first_message_id = str(uuid4())
+            context_id = str(uuid4())  # This will be our session identifier
+
+            send_message_payload_1 = {
+                "message": {
+                    "role": "user",
+                    "parts": [
+                        {
+                            "kind": "text",
+                            "text": "My name is Alice and I work as a software engineer.",
+                        }
+                    ],
+                    "messageId": first_message_id,
+                    "contextId": context_id,  # Link messages to same conversation
+                },
+            }
+
+            request_1 = SendMessageRequest(
+                id=str(uuid4()), params=MessageSendParams(**send_message_payload_1)
+            )
+            response_1 = await client.send_message(request_1)
+
+            assert response_1 is not None
+            result = TestResult.model_validate_json(
+                response_1.root.result.status.message.parts[0].root.text
+            )
+            assert result.name == "Alice"
+            assert result.job.lower() == "software engineer"
+
+            send_message_payload_2 = {
+                "message": {
+                    "role": "user",
+                    "parts": [
+                        {
+                            "kind": "text",
+                            "text": "What's my name, what do I do for work, and what's my age? Let me know if you need more information.",
+                        }
+                    ],
+                    "messageId": str(uuid4()),
+                    "contextId": response_1.root.result.contextId,  # Same context to continue conversation
+                    "taskId": response_1.root.result.id,
+                },
+            }
+
+            request_2 = SendMessageRequest(
+                id=str(uuid4()), params=MessageSendParams(**send_message_payload_2)
+            )
+            response_2 = await client.send_message(request_2)
+
+            assert response_2 is not None
+            result = TestResult.model_validate_json(
+                response_2.root.result.status.message.parts[0].root.text
+            )
+            assert result.name == "Alice"
+            assert result.job.lower() == "software engineer"
+            assert result.age is None
+            assert response_2.root.result.status.state == TaskState.input_required
+
+            # Send a message to the agent to give the age
+            send_message_payload_3 = {
+                "message": {
+                    "role": "user",
+                    "parts": [{"kind": "text", "text": "My age is 30."}],
+                    "messageId": str(uuid4()),
+                    "contextId": response_1.root.result.contextId,  # Same context to continue conversation
+                    "taskId": response_1.root.result.id,
+                },
+            }
+            request_3 = SendMessageRequest(
```

Comment:
Is it worth adding an extra step to test "overriding" information? Like. "ups forget what I said, I turned 31 last week"

Reply:
Hmm. I don't think so? At that point its more testing out how the model responds to clarifying/changing info about the history, which seems more related to model performance and not as related to whether we are correctly giving it the past history. Let me know if you think the overriding info adds something necessary to testing whether or not the history is provided and the state managed.

Reply:
makes sense